src/hooks/useCancellationType.ts

```diff
@@ -0,0 +1,42 @@
+import {useEffect, useMemo, useRef, useState} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import type {CancellationType} from '@src/CONST';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+function useCancellationType(): CancellationType | undefined {
+    const [cancellationDetails] = useOnyx(ONYXKEYS.NVP_PRIVATE_CANCELLATION_DETAILS);
+
+    const [cancellationType, setCancellationType] = useState<CancellationType | undefined>();
+
+    const previousCancellationDetails = useRef(cancellationDetails);
```

Comment:
At the first glance this looked a bit complex to me. Can we add a comment on why we need the ref? Also in useMemo do we need to update the `ref.current` ?

Reply:
Added a comment